synapse/storage/databases/main/event_federation.py

```diff
@@ -207,6 +219,18 @@ def _get_auth_chain_difference_using_cover_index_txn(
                 seen_chains.add(chain_id)
                 chain_to_event.setdefault(chain_id, {})[sequence_number] = event_id
 
+        # Check that we actually have a chain ID for all the events.
+        events_missing_chain_info = initial_events.difference(chain_info)
```

Comment:
I was going to suggest using `initial_events >= chain_info` as we don't care about the difference, just if there is one. But `chain_info` is a `dict` and it would probably be more expensive to convert it to a set first. 🤷

Reply:
:shrug: